### ISSUE TITLE
QuantileLoss with every Llst not containing the 0.5 Quantile

### ISSUE THREAD

**Issue by Beich2010** [jdb78/pytorch-forecasting#1348]
- PyTorch-Forecasting version: 1.0.0
- PyTorch version: 2.0.1
- Python version: 3.10.0
- Operating System: Ubuntu 22.04.2 LTS

Hey there,

When using the Quantile Loss, I noticed a little something. Either the documentation should be changed or the desired behavior should be created by a small change in the code. If necessary, I can also create a pull request.

Thank you in advance.

### Expected behavior

Using the Quantile Loss only to get one Point Forecast, or to get a forecast of a list of Quantiles not containing the median.

Documentation of [Quantile Loss](https://pytorch-forecasting.readthedocs.io/en/stable/api/pytorch_forecasting.metrics.quantile.QuantileLoss.html) says:

quantiles – quantiles for metric 

and does not specify this constraint. 

I executed the code below in order to a point forecast of the 0.9 Quantile and expected to get the quantile as output. This is further reproducible with every list of quantiles not containing the 0.5 quantile. I.e. quantiles = [0.1,0.8,0.9].

### Actual behavior

Error in:
 pytorch_forecasting/metrics/quantile.py", line 50, in to_prediction
    idx = self.quantiles.index(0.5)

As the index 0.5 not exists it throws this Value Error.

The error comes from QuantileLoss the to_prediction method
```
    def to_prediction(self, y_pred: torch.Tensor) -> torch.Tensor:
        """
        Convert network prediction into a point prediction.

        Args:
            y_pred: prediction output of network

        Returns:
            torch.Tensor: point prediction
        """
        if y_pred.ndim == 3:
            idx = self.quantiles.index(0.5)       <---------  Here
            y_pred = y_pred[..., idx]
        return y_pred
```

### Code to reproduce the problem

```
import os
import warnings

warnings.filterwarnings("ignore")

os.chdir("../../..")

import lightning.pytorch as pl
from lightning.pytorch.callbacks import EarlyStopping
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import torch

from pytorch_forecasting import NHiTS, TimeSeriesDataSet
from pytorch_forecasting.data import NaNLabelEncoder
from pytorch_forecasting.data.examples import generate_ar_data
from pytorch_forecasting.metrics import  QuantileLoss

data = generate_ar_data(seasonality=10.0, timesteps=400, n_series=100, seed=42)
data["static"] = 2
data["date"] = pd.Timestamp("2020-01-01") + pd.to_timedelta(data.time_idx, "D")
data.head()

# create dataset and dataloaders
max_encoder_length = 60
max_prediction_length = 20

training_cutoff = data["time_idx"].max() - max_prediction_length

context_length = max_encoder_length
prediction_length = max_prediction_length

training = TimeSeriesDataSet(
    data[lambda x: x.time_idx <= training_cutoff],
    time_idx="time_idx",
    target="value",
    categorical_encoders={"series": NaNLabelEncoder().fit(data.series)},
    group_ids=["series"],
    # only unknown variable is "value" - and N-HiTS can also not take any additional variables
    time_varying_unknown_reals=["value"],
    max_encoder_length=context_length,
    max_prediction_length=prediction_length,
)

validation = TimeSeriesDataSet.from_dataset(training, data, min_prediction_idx=training_cutoff + 1)
batch_size = 128
train_dataloader = training.to_dataloader(train=True, batch_size=batch_size, num_workers=0)
val_dataloader = validation.to_dataloader(train=False, batch_size=batch_size, num_workers=0)


pl.seed_everything(42)
trainer = pl.Trainer(accelerator="cpu", gradient_clip_val=0.1)
net = NHiTS.from_dataset(
    training,
    learning_rate=3e-2,
    weight_decay=1e-2,
    loss=QuantileLoss(quantiles=[0.9]),
    backcast_loss_ratio=0.0,
    hidden_size=64,
    optimizer="AdamW",
)

trainer.fit(
            net,
            train_dataloaders=train_dataloader,
            val_dataloaders=val_dataloader,
)
```
